Automattic/wp-calypso
# New Plugin Details page

### Summary 
Create a Plugin Details page with a new layout as oU6nEIeiKSg8CayhEtgR6b-fi-6559%3A49151
The implementation was made with a new component to be shown when the `marketplace` feature flag is enabled, this component is based on the old `plugin` component, but created as a functional React component.

### Testing instructions
1. Go to plugins page
2. Click in one of the plugins, this should redirect you to the plugin details page.
3. Once on the plugin details page, verify:
- The layout matches with oU6nEIeiKSg8CayhEtgR6b-fi-6559%3A49151
- The mobile version is working as expected
- The CTA actions redirect to a new page when installing and show a thank you page at the end.
- Check the CTA for the 3 possibilities:
  - Install when atomic
  - Upgrade to business and install
  - Install when not atomic

### Screenshots
|Before | After| After Mobile|
|-------|------|------|
|<img width="720" alt="Screen Shot 2021-11-07 at 11 41 09" src="https://user-images.githubusercontent.com/5039531/140652107-985f621d-194e-40ad-adb3-3b578ec8db77.png">|<img width="1188" alt="Screen Shot 2021-11-07 at 11 38 49" src="https://user-images.githubusercontent.com/5039531/140652115-cab3139a-ef5b-483b-9f18-49409743d467.png">|<img width="483" alt="Screen Shot 2021-11-07 at 11 39 42" src="https://user-images.githubusercontent.com/5039531/140652120-18968e17-d3c3-443f-9b1f-b661bf6e88cb.png">|


### Checklist

- [x] Turn a class component into a functional component
- [x] New Layout (wider, 2 columns)
- [x] Header with summary 
- [x] Description / FAQ's
- [x] Include cover image on the description
- [x] Remove the Read more action
- [x] Secondary Info
- [x] Add site list section
- [x] Fix the mobile version
- [x] CTA: Install when atomic
- [x] CTA: Upgrade to business and install
- [x] CTA: Install when not atomic

On separate / next PRs
- [ ] Add placeholder page
- [ ] Add site list styling
- [ ] Add star icons to rating
- [ ] Make sure that author link redirects to our plugin search results showing plugins of that author eg for yoast `https://api.wordpress.org/plugins/info/1.2/?action=query_plugins&request[author]=yoast`

---


Fixes #55577
